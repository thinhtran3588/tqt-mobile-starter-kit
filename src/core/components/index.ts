export * from '@tqt/mobile/lib/components';
export * from './icon/icon.component';
export {
  Provider as PaperProvider,
  DefaultTheme as PaperDefaultTheme,
  Colors,
  Button,
  Appbar,
  Card,
  ActivityIndicator,
  Avatar,
  Badge,
} from 'react-native-paper';
export {StatusBar, Alert, ScrollView} from 'react-native';
