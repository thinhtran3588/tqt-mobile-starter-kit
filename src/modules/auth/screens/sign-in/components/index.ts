export * from './social-sign-in/social-sign-in.component';
