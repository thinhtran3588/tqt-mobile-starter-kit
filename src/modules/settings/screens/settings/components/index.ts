export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
