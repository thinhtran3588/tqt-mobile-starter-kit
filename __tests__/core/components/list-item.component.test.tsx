import React from 'react';
import {render, toJSON} from '@test-utils';
import {ListItem} from '@core/components';

it('renders correctly', async () => {
  const {baseElement} = render(<ListItem title='List Item' />);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders with icons correctly', async () => {
  const {baseElement} = render(<ListItem title='List Item' leftIcon='format-color-fill' rightIcon='chevron-right' />);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders with switch correctly', async () => {
  const {baseElement} = render(
    <ListItem title='List Item' leftIcon='format-color-fill' switchRight switchRightValue />,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});
