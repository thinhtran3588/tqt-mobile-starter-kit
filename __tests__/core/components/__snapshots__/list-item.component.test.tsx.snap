// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(0, 0, 0, 0.1)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              List Item
            </Text>
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders with icons correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(0, 0, 0, 0.1)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰉦
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              List Item
            </Text>
          </View>
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅂
            </Text>
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders with switch correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(0, 0, 0, 0.1)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰉦
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              List Item
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {},
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingRight": 10,
                },
              ]
            }
          >
            <Switch
              trackColor={
                Object {
                  "false": "",
                  "true": "#03dac4",
                }
              }
              value={true}
            />
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
